transport/http/client.go

```diff
@@ -5,12 +5,14 @@ import (
 	"context"
 	"encoding/json"
 	"encoding/xml"
+	"errors"
 	"io"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 
 	"github.com/go-kit/kit/endpoint"
+	"github.com/golang/protobuf/proto"
```

Comment:
Do we want to add this dependency here? I would venture a guess that `github.com/go-kit/kit/transport/http` is one of the most commonly used packages within Go kit and I would also venture a guess that pairing it with protobuf does not account for the majority of use cases.

Is it possible to offer the protobuf functionality as a separate package (I don't know as I don't have time to review the rest of this PR thoroughly at the moment)?

To be clear, my objection is not strenuous. Although `github.com/golang/protobuf/proto` is a large package, it doesn't bring with it any transitive dependencies outside the standard library. But this would be the first non-Go kit and non-standard library dependency here, so I think it's worth being conservative if possible.

Reply:
I had the same hesitation, for the record.

Reply:
Agreed... I'd love to have this contribution, but I see this more in a subpackage of `transport/http`.

Reply:
I'm down to push these to a subpackage. Any recommendations on package name? `proto`? `httpproto`?